base/version_comparable.go

```diff
@@ -129,10 +135,21 @@ func (a *ComparableVersion) Less(b *ComparableVersion) bool {
 	return false
 }
 
-func (pv *ComparableVersion) String() string {
-	pv.strOnce.Do(func() {
-		pv.str = pv.formatComparableVersion()
-	})
+// AtLeastMinor returns true there is a major or minor downgrade.
+func (a *ComparableVersion) AtLeastMinorDowngrade(b *ComparableVersion) bool {
+	if a.epoch > b.epoch {
+		return true
+	}
+	if a.major > b.major {
+		return true
+	}
+	if a.minor > b.minor {
+		return true
+	}
+	return false
```

Comment:
I think there's a bug here - this would return true for 3.2.0 -> 4.1.0, wouldn't it?

Reply:
whoops